packages/cli/src/controller/migrate/abis/migrate-abis.controller.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import fs from 'fs';
+import path from 'path';
+import {uniqWith} from 'lodash';
+import {DEFAULT_ABI_DIR} from '../../generate-controller';
+import {SubgraphDataSource, SubgraphProject} from '../types';
+
+function extractAllAbiFiles(dataSources: SubgraphDataSource[]): string[] {
+  return uniqWith(
+    dataSources.flatMap((dataSource) =>
+      dataSource.mapping.abis.map((abi) => abi.file).filter((file) => file !== null && file !== undefined)
+    )
+  );
+}
+
+async function copyAbiFilesToTargetPathAsync(abiFiles: string[], targetPath: string): Promise<void> {
+  try {
+    for (const file of abiFiles) {
+      const fileName = path.basename(file);
+      const targetFilePath = path.join(targetPath, fileName);
+      await fs.promises.copyFile(file, targetFilePath);
+    }
+    console.log('ABI files copied successfully.');
+  } catch (error) {
+    console.error('Error copying ABI files:', error);
```

Comment:
Should this throw and fail?

Reply:
I think we should allow this pass but log the error, so in this way cli will not remove the migrated project directory,  this allow user to manually fix/copy over the abi files.